docs/index.rst

```diff
@@ -10,51 +10,102 @@
+* block :  a data element in a TNG trajectory containing a specific type of data eg. positions, velocities or box vectors.
+* block id : an integer (a long long) that indicates the type of data contained in a block.
+* block name : a name that matches a specific block id, normally starts with the TNG prefix.
+* particle dependency : indicates whether the data in a block is dependent on the particles in the simulation, (eg positions) or is not (eg box vectors).
+
+Notes on the TNG format and PyTNG
+=================================
+
+While the TNG format supports storage of simulations conducted in the
+grand canonical ensemble, PyTNG does not currently support this. Additonally,
+the TNG format includes a TNG_MOLECULES block that contains the simulation
+topology. PyTNG does not currently make use of this information.
 
-TNG files can be read using the TNGFile class as a file handle,
+
+Usage example for TNGFileIterator
+=================================
+
+TNG files can be read using the TNGFileIterator class as a file handle,
 which supports use as a context manager.
-The TNGFile returns one frame at a time, which each frame being returned as a
-namedtuple with the following attributes:
 
-============== ======== =======================================
-attribute      type     description
-============== ======== =======================================
-``positions``  float32  coordinates for each atom in the frame
-``time``       float    current system time
-``step``       int      frame index
-``box``        float32  3x3 matrix of the system volume
-============== ======== =======================================
+The TNGFileIterator has attributes related to the trajectory metadata, such as
+the number of integrator steps, the number of steps with data, the block_ids 
+available at each step, and the stride at which each block is written.
+
+The TNGFileIterator returns one frame at a time, which is accessed from the
+:attr:`TNGFileIterator.current_integrator_step` attribute or as part of any
+slicing or indexing operation. A NumPy array of the right size must then be
+provided to a getter method for the data to be read into.
+
+An example of how to read positions from a TNG file is shown below.
 
-For example, the coordinate information can be accessed via the `.positions` 
-attribute of the returned object.
 
 .. code-block:: python
 
-  import pytng
+   import pytng
+   import numpy as np
+
+   with pytng.TNGFileIterator("traj.tng", 'r') as tng:
+
+      # make a numpy array to hold the data using helper function
+      # this array will then be updated in-place 
+      
+      positions = tng.make_ndarray_for_block_from_name("TNG_TRAJ_POSITIONS")
+      
+      # the TNG API uses regular strides for data deposition, here we stride
+      # over the whole trajectory for the frames that have position data
+      # len(tng) is the total number of steps in the file.
+
+      for ts in tng[0:len(tng):tng.block_strides["TNG_TRAJ_POSITIONS"]]:
```

Comment:
Just a question: is there a way to iterate over all steps that contain any data, e.g., if block_strides for positions are 100 and 1000 for velocities then we would automatically do stride 100 and for 9 out of 10 steps, we just don't get velocity data.

More weirdly, when positions are saved every 5 steps and velocities every 11 steps then we would get only a few steps that have both (at step = n * 5 * 11, n = 0, 1, ,2, ...).

Reply:
The easiest solution I thought of was to stride over the greatest common divisor of the strides.
This is already computed as `TngFileIterator._gcd`, but I haven't made use of it yet.

Does that sound reasonable?

Reply:
The GCD(5, 11) = 1 so what does this mean in terms of iterating? We get a lot of empty steps?

Reply:
For the likely more common case of 10,100 the GCD sounds pretty sane, i.e., when GCD(m, n) == m or n. In the case where we get empty steps then the calling code will need to decide what to do (such as not returning a step) and that will be rather costly.

I'd much prefer if TNGFileIterator never returned empty frames. It tags frames so user code will just have to work with the frame metadata.

From MDAnalysis' perspective I also want to know what `len(tng)` should mean; I think the only sensible interpretation is "total number of non-empty frames" (like TRR).

Any opinions @richardjgowers @jbarnoud @kain88-de ?

Reply:
Working with steps is closer to the way the format itself works. My suggestion is that working in the original units of the format is better, we can then provide the user the tools they need to iterate with the minimum of wasted steps (obviously with big improvements in docs required).

I am happy to change it up though! No dramas.

I see where your coming from in regards to the length, would this mean the total number of frames with any block attached would become the length? This would still result in blank reads although a lot fewer. The other option is to have a separate length in each block "dimension".

Open to suggestion!

Reply:
Hey @orbeckst I have tried to clear this up a bit and add more info in the docs.

For blank reads we currently fill the array with np.nan. Not the most elegant solution but gets the point across.  I have added an example that demonstrates this to the docs.

Reply:
Is processing np.nan expensive? 

In any case, I don't have a better idea right away (I guess we can't temporarily set the whole array to NULL or None?) Is there a way for calling code to quickly learn what data are available – is it just `np.all(np.isnan())`, which would be expensive? Could you add an example that shows how calling code can find out efficiently which data it can pull out?

Reply:
The example above is for the simple case when the strides coincide. That's good to introduce the basics.

Now add an example where the strides differ. If I don't know how to do this I have no idea how write a Reader for e.g. MDA.